### PR TITLE
doc: update FIPS provider version information

### PR DESCRIPTION
**Pull request by paulidale**
With 3.0.8 validated, we need to note this in the documentation.

Counterpart of #21049 for 3.0.x

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Second review required.

**Review by levitte — APPROVED** [review]

File: doc/man7/OSSL_PROVIDER-FIPS.pod
Comment (t8m) [inline doc/man7/OSSL_PROVIDER-FIPS.pod:440]: I would just remove this.

File: doc/man7/fips_module.pod
Comment (t8m) [inline doc/man7/fips_module.pod:482]: And this.

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by paulidale** [comment]
Ping @openssl/committers for second review please.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the reviews.

**Pull request closed without merging**
